enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestionResult.tsx

```diff
@@ -70,14 +69,15 @@ export const InteractiveQuestionResult = ({
   const [questionView, setQuestionView] =
     useState<QuestionView>("visualization");
 
-  const { isQuestionLoading } = useInteractiveQuestionContext();
+  const { question, queryResults, isQuestionLoading } =
```

Comment:
Will we have issues bringing this data back to `context`? One of the things I'm thinking about is that if, for example, when we apply a filter, the _whole_ component  (i.e. everything under the context) will reload. It might be jarring if someone has a custom layout and they watch their whole component reload every time they make a change

Reply:
@oisincoveney good point, let me do some re-rendering checks to see how often the child re-renders first.

In the worst-case scenario, we could re-implement this in Redux: create a random instance identifier on component mount (e.g. a nanoid), then create a hashmap of instanceId to question state.

Reply:
PS. I just noticed that the entire component goes into a loading state when a filter changes, as opposed to just the visualization. I think this can be fixed by having a separate loading state for question loading (first mount), versus query running. Let me fix that.

Reply:
I measured the component re-rendering and it doesn't seem to be different from `master`, so I think it's safe to continue with this. We can consider converting this to Redux to optimize when a need arises.

Reply:
Ok perfect!